ibis/backends/base/sql/alchemy/__init__.py

```diff
@@ -43,30 +54,369 @@
 )
 
 
+class _AutoCloseCursor:
+    """
+    Wraps a SQLAlchemy ResultProxy and ensures that .close() is called on
+    garbage collection
+    """
+
+    def __init__(self, original_cursor):
+        self.original_cursor = original_cursor
+
+    def __del__(self):
+        self.original_cursor.close()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, type, value, tb):
+        self.original_cursor.close()
+
+    def fetchall(self):
+        return self.original_cursor.fetchall()
+
+
 class BaseAlchemyBackend(BaseSQLBackend):
     """
     Base backend class for backends that compile to SQL with SQLAlchemy.
     """
 
     database_class = AlchemyDatabase
     table_class = AlchemyTable
+    compiler = AlchemyCompiler
+    has_attachment = False
```

Comment:
What does this mean? Is this referring to the SQLite `ATTACH` statement?

Reply:
I had a look, this is only used when inserting data in SQLite (`insert` and `load_data`). When `attach` in SQLite has been called, this add the parameter `DataFrame.to_sql(schema=current_database)`.